Toric/Mathlib/Geometry/Convex/Cone/PolyhedralCone.lean

```diff
@@ -0,0 +1,225 @@
+/-
+Copyright (c) 2025 Justus Springer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Justus Springer
+-/
+import Toric.Mathlib.Geometry.Convex.Cone.Dual
+import Toric.Mathlib.Algebra.Order.Nonneg.Module
+
+/-!
+
+# Polyhedral cones
+
+Given a bilinear pairing `p` between two `R`-modules `M` and `N`, we define
+polyhedral cones to be pointed cones in `N` that are the dual of a finite set
+in `M` (this means they are the intersection of finitely many halfspaces).
+
+The main statement is that if both `M` and `N` are finite and the pairing is injective
+in both arguments, then polyhedral cones are precisely the finitely generated cones, see
+`IsPolyhedral_iff_fg`. Moreover, we obtain that the dual of a polyhedral cone is again polyhedral
+(`IsPolyhedral_dual_of_IsPolyhedral`) and that the double dual of a polyhedral cone is the
+cone itself (`IsPolyhedral_dual_dual`).
+
+-/
+
+
+namespace PointedCone
+
+open Submodule hiding span
+
+section PartialOrder
+
+variable {R M N : Type*} [CommRing R] [PartialOrder R] [IsOrderedRing R] [AddCommGroup M]
+  [AddCommGroup N] [Module R M] [Module R N] {p : M →ₗ[R] N →ₗ[R] R}
+
+variable (p) in
+def IsPolyhedral (c : PointedCone R N) : Prop :=
+  ∃ t : Finset M, c = dual' p t
+
+lemma IsPolyhedral_def {c : PointedCone R N} :
+    IsPolyhedral p c ↔ ∃ t, t.Finite ∧ c = dual' p t :=
+  ⟨fun ⟨t, ht⟩ => ⟨t, t.finite_toSet, ht⟩,
+   fun ⟨_, ht1, ht2⟩ =>⟨ht1.toFinset, ht1.coe_toFinset.symm ▸ ht2⟩⟩
+
+lemma IsPolyhedral_dual_of_Finite {t : Set M} (h : t.Finite) :
+    IsPolyhedral p (dual' p t) := ⟨h.toFinset, by simp⟩
```

Comment:
```suggestion
lemma isPolyhedral_dual_of_finite {t : Set M} (h : t.Finite) :
    IsPolyhedral p (dual' p t) := ⟨h.toFinset, by simp⟩
```
similarly below. The rule is that `UpperCamelCase` becomes `lowerCamelCase` in lemma names

Reply:
Corrected this and similar lemma names below. I hope all names comply with the rule now.